stop_and_wait_plugin/src/stop_and_wait_plugin.cpp

```diff
@@ -345,7 +353,7 @@ std::vector<carma_planning_msgs::msg::TrajectoryPlanPoint> StopandWait::compose_
 
   std::vector<double> filtered_speeds = basic_autonomy::smoothing::moving_average_filter(speeds, config_.moving_average_window_size);
 
-  for (size_t i = 0; i < filtered_speeds.size(); i++)
+  for (size_t i = 1; i < filtered_speeds.size(); i++)
```

Comment:
Why has this index changed

Reply:
This is to let the keep the first velocity, which is the vehicle's current velocity. We shouldn't force the vehicle's first velocity to be 0 because plugins supposed to plan starting from the last available speed to be continuous.

Reply:
So the correct steps after keeping the first velocity is to calculate the dx it takes to slow down to 0 and keep that distance in the trajectory as well. This is same behavior we discussed. Otherwise due to the 0 velocity, dx will be 0, and again we will have the original problem of zigzag. Because despite using 0 velocity here to calculate the dx, the later components still uses the first non-zero first velocity to convert back, which conflicts. So this dx calculation is handled in `is_first_extrapolation_point` now. And then after that dx, the speed is 0, so the dx can be safely 0.